resources/images/s3/filestorage.go

```diff
@@ -204,13 +222,36 @@ func (s *SimpleStorageService) UploadArtifact(ctx context.Context,
 			"Artifact upload failed with HTTP status %v", resp.Status)
 	}
 
+	if id := identity.FromContext(ctx); len(id.Tenant) > 0 && s.setTenantTag {
```

Comment:
consider erroring out when 'set tenant tag' is set and there's no 'tenant' in the jwt